### PR TITLE
Add html merged renderer

### PR DESCRIPTION
**Pull request by DigiLive**
Adds an renderer which shows merge of version1 and version2.

The name "Merged" might have to change to a more suitable name.

### CODE REVIEW

**Comment by DigiLive** [comment]
Please don't merge the PR yet.
The pushed commits are to discuss the output of the new renderer at first.

File: example/example.php
Comment (JBlond) [inline example/example.php:5]: Please add a line break

**Review by JBlond — CHANGES_REQUESTED** [review]

File: example/example.php
Comment (DigiLive) [reply example/example.php]: Sorry about that. It seems the IDE doesn't properly format the code when it has mixed languages.

**Review by JBlond — APPROVED** [review]

**Review by JBlond — APPROVED** [review]

File: lib/jblond/Diff/Renderer/Html/Merged.php
Comment (JBlond) [inline lib/jblond/Diff/Renderer/Html/Merged.php:38]: Here is the PHP doc missing

Comment (DigiLive) [reply lib/jblond/Diff/Renderer/Html/Merged.php:38]: Actually there's a bit more missing or invalid. I intended to fix all when the acutal output of the renderer was discussed.

**Comment by DigiLive** [comment]
Do you have any remarks on the output?
afbeelding
Deleted lines are shown when hovering the lines which are marked with an `*` (Line 8).
Maybe there's a better solution to indicate removed lines?

**Comment by JBlond** [comment]
I like the output. The hover effect is good. The star should be lager. I overlooked it, if you hadn't mentioned it.

**Comment by DigiLive** [comment]
Improved visibility of removed lines...
afbeelding

**Comment by DigiLive** [comment]
It seems to be compatible with the new marking levels.

But about the naming of the renderers... I'm a bit confused.
1. HTML Side by Side Diff. Seems ok.
2. HTML Inline Diff. Isn't this one a unified diff, but with inline markings?
3. HTML Unified Diff. Seems ok, but just another style of whats currently named HTML Inline Diff?
4. HTML Merged Diff. Isn't this one just an inline diff?

**Comment by JBlond** [comment]
By definition line diff is the highlighting of the changed characters. A "plain" diff would is like "Inline Marking" level none.

**Comment by JBlond** [comment]
One unit test fails
```
Tests\Diff\Renderer\Html\HtmlRenderersTest::testMerged
Failed asserting that two strings are equal.
php-diff\tests\Diff\Renderer\Html\HtmlRenderersTest.php:110
```

**Comment by DigiLive** [comment]
> One unit test fails

Darn... I've fixed it before, but I had to include the changes from the last PR which reintroduced this failure.
It's fixed at a following commit.


> By definition line diff is the highlighting of the changed characters. A "plain" diff would is like "Inline Marking" level none.

Ok, but what we call HTML Inline Diff now, seems like a unified diff to me.
afbeelding

**Comment by DigiLive** [comment]
Regarding the renderer names, I've created issue #77 
After reviewing and approval, I think this PR could be merged.

**Review by JBlond — APPROVED** [review]

**Comment by JBlond** [comment]
For backward compatibility the names should remain. However in v3 we could change that and there is also the jump to the next PHP version.

**Pull request merged**
